streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredTimestampWindowStoreTest.java

```diff
@@ -89,4 +92,49 @@ public void shouldNotExceptionIfFetchReturnsNull() {
         assertNull(store.fetch("a", 0));
     }
 
+    @Test
+    public void shouldNotThrowExceptionIfSerdesCorrectlySetFromProcessorContext() {
+        EasyMock.expect(innerStoreMock.name()).andStubReturn("mocked-store");
+        EasyMock.replay(innerStoreMock);
+        final MeteredTimestampedWindowStore<String, Long> store = new MeteredTimestampedWindowStore<>(
+            innerStoreMock,
+            10L, // any size
+            "scope",
+            new MockTime(),
+            null,
+            null
+        );
+        store.init(context, innerStoreMock);
+
+        try {
+            store.put("key", ValueAndTimestamp.make(42L, 60000));
+        } catch (final StreamsException exception) {
```

Comment:
I don't understand this ? Why do we catch `StreamsException`? Should the test not just fail if an exception is thrown?

Reply:
To distinguish between a test failure and a test that is terminated by an exception. If a `StreamsException` that  caused by a `ClassCastException` was thrown by `store.put()`, then the test failed because the behaviour under test was not correct, otherwise it was terminated due to some other issues.